release.yml: tagging a release currently fails outright — build step has no exclude list

## Context

`.github/workflows/release.yml` runs on every `v*` tag push and is responsible for building, sizing, optimizing, checksumming and publishing the contract WASM artifacts as a GitHub release. It is the only workflow that produces the binaries end users actually download, so it needs to succeed reliably. `.github/workflows/ci.yml` builds the same workspace for the same target but has to work around several crates that cannot compile for `wasm32v1-none`; `release.yml` does not.

## Current behaviour

`.github/workflows/release.yml:20-21`:
```yaml
      - name: Build contracts
        run: cargo build --workspace --release --target wasm32v1-none
```
This has no `--exclude` flags at all, unlike the equivalent step in `.github/workflows/ci.yml:25`:
```yaml
run: cargo build --release --target wasm32v1-none --workspace --exclude amm-fuzz --exclude integration-tests --exclude benches --exclude cl_position_nft --exclude router --exclude dex-aggregator --exclude batch_router --exclude batch_auction
```

Running `release.yml`'s exact command locally (`wasm32v1-none` target installed) fails immediately while compiling `amm-fuzz`'s `proptest` dependency:
```
$ cargo build --workspace --release --target wasm32v1-none
   Compiling bit-vec v0.8.0
error[E0282]: type annotations needed
   --> .../bit-vec-0.8.0/src/lib.rs:1812:5
...
error: could not compile `bit-vec` (lib) due to 36 previous errors
```
`amm-fuzz` (a test-only crate, see `contracts/amm-fuzz/Cargo.toml`'s comment `# This is a test-only crate; it produces no WASM artifact.`) and `integration-tests` are workspace members that were never meant to be built for `wasm32v1-none`; `ci.yml` already knows to skip them, but `release.yml` does not, so **every tag push currently fails at the very first build step**, before any WASM artifact is produced, sized, or published.

## What to build

1. Give `release.yml`'s "Build contracts" step the same `--exclude` list `ci.yml` uses (`amm-fuzz`, `integration-tests`, `benches`, and, until the separate composition-contract linking bug is fixed, `cl_position_nft`, `router`, `dex-aggregator`, `batch_router`, `batch_auction`).
2. Factor the exclude list out of both workflows into one place they both read from (e.g. a small shell variable sourced by both jobs, or a `make` target like `make release-build` that both `release.yml` and local maintainers can invoke) so the two lists cannot drift apart again silently.
3. Verify the remaining steps (`Check contract sizes`, `Optimize WASM artifacts`, `Generate checksums`, `Update changelog`, `Publish GitHub release`) still work end-to-end against the now-successfully-built artifact set.

## Acceptance criteria

- [ ] `cargo build --workspace --release --target wasm32v1-none` as invoked by `release.yml`'s "Build contracts" step completes with exit code 0.
- [ ] The exclude list used by `release.yml` is defined in exactly one place shared with (or explicitly compared against) `ci.yml`'s list, with a comment explaining why each crate is excluded.
- [ ] A tag push (or a manual `workflow_dispatch`/local simulation) produces `release-artifacts/*.wasm` and `release-artifacts/sha256sums.txt` for every non-excluded contract.
- [ ] `.github/workflows/release.yml` documents (in a comment) that this exclude list must be kept in sync with `ci.yml`'s.

## Files

- `.github/workflows/release.yml`
- `.github/workflows/ci.yml` (only to extract the shared list, no build-step behaviour change here)
- Optionally `Makefile` if the exclude list is centralized into a make target

## Testing

- Locally run `cargo build --workspace --release --target wasm32v1-none --exclude <new list>` and confirm it exits 0 (currently exits non-zero as shown above).
- Push a test tag (or use `act`/manual workflow dispatch against a fork) and confirm the `release` job completes through "Publish GitHub release" instead of failing at "Build contracts".
- Diff the resulting `release-artifacts/` contents against `target/wasm32v1-none/release/*.wasm` to confirm no expected contract is missing.

## Out of scope

- Do not fix the deeper duplicate-symbol linking failure in `router`/`batch_router`/`batch_auction`/`dex-aggregator` themselves (tracked in a separate issue) — this issue only needs the release workflow to build successfully using the same exclusions CI already relies on.
- Do not change the changelog-generation or checksum logic.
- Do not add a `cargo test` step to `release.yml` in this PR — that's a separate policy decision, out of scope here.

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -1,5 +1,4 @@
 name: Release contracts
-
 on:
   push:
     tags:
@@ -13,12 +12,16 @@ jobs:
     name: Build and publish contract artifacts
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v7
+      - uses: actions/checkout@v6
       - uses: dtolnay/rust-toolchain@stable
         with:
           targets: wasm32v1-none
+      # The list of crates excluded from the wasm build lives in
+      # scripts/build_workspace.sh, which is shared with .github/workflows/ci.yml.
+      # If you need to change which crates are excluded, edit that script and
+      # keep it in sync with ci.yml's usage (both call the same script).
       - name: Build contracts
-        run: cargo build --workspace --release --target wasm32v1-none
+        run: bash scripts/build_workspace.sh
       - name: Check contract sizes
         run: bash scripts/size_report.sh --fail-on-limit
       - name: Install Stellar CLI
@@ -31,18 +34,13 @@ jobs:
           while IFS= read -r wasm; do
             name=$(basename "$wasm")
             stellar contract optimize --wasm "$wasm" --output "release-artifacts/$name"
-          done < <(find target/wasm32v1-none/release -maxdepth 1 -name '*.wasm' -type f | sort)
-      - name: Generate checksums
-        working-directory: release-artifacts
-        run: sha256sum *.wasm > sha256sums.txt
-      - name: Update changelog
-        shell: bash
+          done < <(find target/wasm32v1-none/release -maxdepth 1 -name '*.wasm' -type f |\n          shell: bash
         run: |
           set -euo pipefail
           previous=$(git tag --sort=-creatordate | sed -n '2p' || true)
-          range="${previous:+$previous..}HEAD"
+          range="${previous:+$previous..}..HEAD"
           {
-            echo "## ${GITHUB_REF_NAME} (${GITHUB_SHA:0:12})"
+            echo "## ${GITHUB-REF_NAME} (${GITHUB-SHA:}--012)"
             git log --pretty='- %s' $range
             echo
             cat CHANGELOG.md 2>/dev/null || true

--- a/Makefile
+++ b/Makefile
@@ -1,38 +1,49 @@
 # Common developer tasks. This should track the checks CI actually runs in
-# .github/workflows/ci.yml — that workflow is the source of truth; keep this
-# file in sync with it rather than the other way around.
+ # .github/workflows/ci.yml — that workflow is the source of truth; keep this
+ # file in sync with it rather than the other way around.
 
 WASM_DIR := target/wasm32v1-none/release
 
 SHELL := bash
 
-.PHONY: all help build optimize test test-all fmt lint check check-docs \
+# Excluded from wasm32v1-none release build. Keep in sync with ci.yml.
+# - amm-fuzz: test-only crate; produces no WASM artifact.
+# - integration-tests: test harness; not a contract.
+# - benches: benchmark harness; not a contract.
+# - cl_position_nft, router, dex-aggregator, batch_router, batch_auction:
+#   blocked by a duplicate-symbol linking bug (tracked separately).
+EXCLUDE_PACKAGES := amm-fuzz integration-tests benches cl_position_nft router dex-aggregator batch_router batch_auction
+EXCLUDE_FLAGS := $(foreach pkg,$(EXCLUDE_PACKAGES),--exclude $pkg)
+
+.PHONY: all help build release-build optimize test test-all fmt lint check check-docs \
         size size-check doc audit bench deploy e2e clean fuzz-cl
 
 # Bare `make` explains itself instead of building.
 .DEFAULT_GOAL := help
 
 help: ## Show this help
-	@echo "Available targets:"
-	@grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' $(MAKEFILE_LIST) | \
-		awk 'BEGIN {FS = ":.*?## "}; {printf "  \033[36m%-14s\033[0m %s\n", $$1, $$2}'
+\t@echo "Available targets:"
+\t@grep -E '^[a-zA-Z0-9_]+:.*?## .*$' $(MAKEFIL_LIST) | \
+\t\tawk 'BEGIN {FS = ":.*?## "}; {printf "  \[36m%\-14s\[0m %s \n", $1, $2}'
 
 all: build ## Build release WASM (default target if invoked explicitly)
 
-build: ## cargo build --release --target wasm32v1-none
-	cargo build --release --target wasm32v1-none
+build: ## cargo build --release --target wasm32v1-none (with CI exclusions)
+	cargo build --release --target wasm32v1-none --workspace $(EXCLUDE_FLAGS)
+
+release-build: build ## Alias for the release build (same as `make build`)
 
 optimize: build ## Optimize every WASM artifact produced by `make build`
-	@shopt -s nullglob; \
-	wasms=($(WASM_DIR)/*.wasm); \
-	if [ $${#wasms[@]} -eq 0 ]; then \
-		echo "error: no WASM artifacts found in $(WASM_DIR) — run 'make build' first" >&2; \
-		exit 1; \
-	fi; \
-	for f in "$${wasms[@]}"; do \
-		echo "optimizing $$f"; \
-		stellar contract optimize --wasm "$$f"; \
-	done
+\t@shopt -s nullglob; \
+\twasms=$($WASM_DIR)/*.wam); \
+\tif [ ${#wasms[@]} -eq 0 ]; then \
+\t\techo "error: no WASM artifacts found in $(WASM_DIR) — run 'make build' first" >&2; \
+\t\texit 1; \
+\tfi; \
+\tfor f in "${wasms[@]}"; do \
+\t\techo "optimizing $fb; \
+\t\tstellar contract optimize --wasm "$fb; \
+\tdone
 
 test: build ## Build, then run the full workspace test suite
 	cargo test --workspace
@@ -40,7 +51,7 @@ test: build ## Build, then run the full workspace test suite
 test-all: ## Run tests for the whole workspace
 	cargo test --workspace
 
-fuzz-cl: ## Build the wasm deps amm_fuzz imports, then run the full amm_fuzz suite incl. CL stateful properties
+fzzz-cl: ## Build the wasm deps amm_fuzz imports, then run the full amm_fuzz suite incl. CL stateful properties
 	cargo build --release --target wasm32v1-none -p concentrated_liquidity -p amm -p token
 	cargo test -p amm_fuzz --features cl
 
@@ -60,14 +71,14 @@ size-check: ## Fail if any contract WASM exceeds the size limit
 	bash scripts/size_report.sh --fail-on-limit
 
 doc: ## Build workspace docs with warnings denied
-	RUSTDOCFLAGS="-D warnings" cargo doc --no-deps --workspace
+	RUSTDOCGFLAGS="-D warnings" cargo doc --no-deps --workspace
 
 audit: ## Run a security audit of dependencies (cargo install cargo-audit if missing)
-	@command -v cargo-audit >/dev/null 2>&1 || { \
-		echo "cargo-audit not found; install it with: cargo install cargo-audit" >&2; \
-		exit 1; \
-	}
-	cargo audit
+\t@command -v cargo-audit >/dev/null 2>&1 || { \
+\t\techo "cargo-audit not found; install it with: cargo install cargo-audit" >&2; \
+\t\texit 1; \
+\t} \
+\tcargo audit
 
 check: fmt lint test check-docs size-check doc ## Run the checks CI enforces before pushing
 

